src/lib/ChargeStation/index.js

```diff
@@ -303,29 +306,60 @@ class Session {
         errorCode: 'NoError',
         status: 'SuspendedEV',
       });
-    } else if (this.options.getCurrentStatus() === 'Charging') {
+    } else if (
+      ['Charging', 'Occupied'].includes(this.options.getCurrentStatus())
+    ) {
       this.kwhElapsed += amountKwhToCharge;
     }
     await sleep(100);
     this.lastMeterValuesTimestamp = this.now();
 
-    await this.options.writeCall('MeterValues', {
-      connectorId: this.connectorId,
-      transactionId: this.transactionId,
-      meterValue: [
-        {
-          timestamp: this.now().toISOString(),
-          sampledValue: [
-            {
-              value: this.kwhElapsed.toFixed(5),
-              context: 'Sample.Periodic',
-              measurand: 'Energy.Active.Import.Register',
-              location: 'Outlet',
-              unit: 'kWh',
-            },
-          ],
+    this.seqNo += 1;
+
+    if (this.ocppVersion === 'ocpp2.0.1') {
+      await this.options.writeCall('TransactionEvent', {
+        eventType: 'Updated',
+        timestamp: this.now().toISOString(),
+        triggerReason: 'MeterValuePeriodic',
+        seqNo: this.seqNo,
+        transactionInfo: {
+          transactionId: this.transactionId,
         },
-      ],
-    });
+        meterValue: [
+          {
+            sampledValue: [
+              {
+                value: Number(this.kwhElapsed.toFixed(3)),
+                context: 'Sample.Periodic',
+                measurand: 'Energy.Active.Import.Register',
+                location: 'Outlet',
+                unitOfMeasure: { unit: 'kWh' },
+              },
+            ],
+            timestamp: this.now().toISOString(),
+          },
+        ],
+        evse: { id: 1, connectorId: this.connectorId },
+      });
+    } else {
+      await this.options.writeCall('MeterValues', {
+        connectorId: this.connectorId,
+        transactionId: this.transactionId,
+        meterValue: [
+          {
+            timestamp: this.now().toISOString(),
+            sampledValue: [
+              {
+                value: this.kwhElapsed.toFixed(5),
+                context: 'Sample.Periodic',
+                measurand: 'Energy.Active.Import.Register',
+                location: 'Outlet',
+                unit: 'kWh',
+              },
+            ],
+          },
+        ],
+      });
+    }
```

Comment:
We should pull this out of ChargeStation index, but we can do it in a separate PR